km_api/know_me/tests/serializers/test_km_user_accessor_serializer.py

```diff
@@ -0,0 +1,31 @@
+from know_me import serializers
+
+
+def test_serialize(
+        api_rf,
+        km_user_factory,
+        km_user_accessor_factory,
+        serializer_context,
+        user_factory):
+    """
+    Test serializing a km_user_accessor.
+    """
+    km_user = km_user_factory()
+    user = user_factory()
+    km_user_accessor = km_user_accessor_factory(
+        km_user=km_user,
+        user_with_access=user)
+
+    serializer = serializers.KMUserAccessorSerializer(
+        km_user_accessor,
+        context=serializer_context)
+
+    expected = {
+        'accepted': False,
+        'can_write': False,
+        'has_private_profile_access': False,
+        'km_user': km_user,
```

Comment:
For `km_user` ait doesn't make sense to return the actual user instance since that is of no use to the client application. I see two options here.

1. We return `km_user` as the user's ID and add another field `km_user_url` to where the user's full details can be retrieved.
2. We return a nested representation similar to `image_resource` for the image content serializer.

Reply:
In order to do #2, would we need to create a KMUserField class in fields.py?

Reply:
That would be the easiest way.

Reply:
**class KMUserField:**
    The field allows know me users to be written using the km user's
    ID, but read using the know me user's serialized form.

**def get_queryset(self):**
        """The queryset returned includes only the know me users owned by
        the Know Me user accessor given as context to the field."""

**def to_internal_value(self, data):**
        """Retrieve the km user with the provided ID."""

**def to_representation(self, value):**
        """Get the serialized representation of the provided km user."""

Reply:
The km_user field is read only so there is no need for a custom field.